### PR TITLE
Support aad sovereign clouds

### LINKED ISSUE #208: Support Azure Active Directory in sovereign cloud
Azure Active Directory has different instances in sovereign clouds:
https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud

And each one has got different endpoint sets, currently the code has hardcoded to the public cloud's endpoint:
https://github.com/appscode/guard/blob/0.3.0/auth/providers/azure/azure.go#L27
https://github.com/appscode/guard/blob/0.3.0/auth/providers/azure/graph/graph.go#L22

Consider introducing a parameter to support different cloud environment for Azure.

### PR DESCRIPTION
**Pull request by karataliu**
Fixes #208

### CODE REVIEW

**Comment by karataliu** [comment]
cc @tamalsaha  @amanohar

**Comment by karataliu** [comment]
ping. thanks.

**Comment by tamalsaha** [comment]
Thanks @karataliu . @amanohar, do you mind taking a look?

File: auth/providers/azure/graph/graph.go [hunk 1]
```diff
@@ -234,7 +232,11 @@ func (u *UserInfo) Name() string {
 }
 
 // New returns a new UserInfo object
-func New(clientID, clientSecret, tenantName string, useGroupUID bool) (*UserInfo, error) {
+func New(clientID, clientSecret, tenantName string, useGroupUID bool, aadEndpoint, msgraphHost string) (*UserInfo, error) {
+	graphEndpoint := "https://" + msgraphHost + "/"
+	baseAPIURL, _ := url.Parse(graphEndpoint + "v1.0")
```
[the remark this answers is not in the document]
Comment (karataliu) [reply auth/providers/azure/graph/graph.go]: Updated. graphEndpoint is fetched from instance metadata, added test for verifying aad endpoint decided environment.

**Comment by karataliu** [comment]
Updated. @amanohar can you take a look

**Comment by karataliu** [comment]
Thanks @amanohar,  @tamalsaha would you please take a look?

**Comment by karataliu** [comment]
ping @tamalsaha

**Comment by karataliu** [comment]
ping @tamalsaha

**Comment by tamalsaha** [comment]
Thanks a lot @karataliu and @amanohar !

**Pull request merged**